x/rollapp/keeper/block_height_to_finalization_queue.go

```diff
@@ -79,15 +74,39 @@ func (k Keeper) GetBlockHeightToFinalizationQueue(
 // RemoveBlockHeightToFinalizationQueue removes a blockHeightToFinalizationQueue from the store
 func (k Keeper) RemoveBlockHeightToFinalizationQueue(
 	ctx sdk.Context,
-	finalizationHeight uint64,
+	creationHeight uint64,
 
 ) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.BlockHeightToFinalizationQueueKeyPrefix))
 	store.Delete(types.BlockHeightToFinalizationQueueKey(
-		finalizationHeight,
+		creationHeight,
 	))
 }
 
+// GetPendingFinalizationQueue returns the blockHeightToFinalizationQueue starting from the input height till the last queue with batches not yet finalized
+func (k Keeper) GetPendingFinalizationQueue(ctx sdk.Context, height uint64) (list []types.BlockHeightToFinalizationQueue) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.BlockHeightToFinalizationQueueKeyPrefix))
+	heightKey := types.BlockHeightToFinalizationQueueKey(height + 1)
+	iterator := sdk.KVStoreReversePrefixIterator(store, heightKey)
+	defer iterator.Close() // nolint: errcheck
+
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.BlockHeightToFinalizationQueue
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		stateInfoIndex := val.FinalizationQueue
+		stateInfo, found := k.GetStateInfo(ctx, stateInfoIndex[0].RollappId, stateInfoIndex[0].Index)
```

Comment:
I mean that this check already happens by the caller
redundant here IMO

Reply:
removed